### PR TITLE
Laravel 10.x Compatibility

### PR DESCRIPTION
**Pull request by laravel-shift**
This is an automated pull request from [Shift](https://laravelshift.com) to update your package code and dependencies to be compatible with Laravel 9.x.

**Before merging**, you need to:

- Checkout the `l10-compatibility` branch
- Review **all** comments for additional changes
- Thoroughly test your package

If you do find an issue, please report it by commenting on this PR to help improve future automation.

### CODE REVIEW

**Comment by laravel-shift** [comment]
:warning: The latest version of Laravel has a dependency for `phpunit/phpunit` of `^9.5.10`. If you would like to also upgrade your tests, you may run the [PHPUnit 9 Shift](https://laravelshift.com/upgrade-phpunit-9) for free.

**Comment by laravel-shift** [comment]
:alembic: **Using this package?** If you would like to help test these changes or believe them to be compatible, you may update your project to reference this branch.

To do so, temporarily add Shift's fork to the `repositories` property of your `composer.json`:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/laravel-shift/laravel-mongodb.git"
        }
    ]
}
```

Then update your dependency constraint to reference this branch:

```json
{
    "require": {
        "jenssegers/mongodb": "dev-l10-compatibility",
    }
}
```

Finally, run: `composer update`

File: composer.json [hunk 1]
```diff
@@ -19,17 +19,17 @@
     ],
     "license": "MIT",
     "require": {
-        "illuminate/support": "^9.0",
-        "illuminate/container": "^9.0",
-        "illuminate/database": "^9.0",
-        "illuminate/events": "^9.0",
-        "mongodb/mongodb": "^1.11"
+        "illuminate/support": "^10.0",
```
Comment (alcaeus) [inline composer.json:22]: Would it make sense to support both Laravel 9.x and 10.x at the same time? I'm not too familiar with the amount of BC breaks that Laravel 10 will bring, but if it's possible to support both without a big hassle, users would profit from that.

File: composer.json [hunk 2]
```diff
@@ -19,17 +19,17 @@
...
     },
     "require-dev": {
-        "phpunit/phpunit": "^9.5.8",
-        "orchestra/testbench": "^7.0",
-        "mockery/mockery": "^1.3.1",
-        "doctrine/dbal": "^2.13.3|^3.1.4"
+        "phpunit/phpunit": "^9.5.10",
+        "orchestra/testbench": "^8.0",
+        "mockery/mockery": "^1.4.4",
+        "doctrine/dbal": "^3.5"
```
Comment (alcaeus) [inline composer.json:32]: Note to self: we should try to get rid of this dependency. With Laravel itself no longer depending on it, we should be able to drop the dependency as well at some point.

Comment (divine) [reply composer.json:32]: We need to get rid of a lot things, library needs a complete revamp even though the new version of Laravel comes soon.

File: .github/workflows/build-ci.yml [hunk 3]
```diff
@@ -41,7 +41,6 @@ jobs:
                     - '4.4'
                     - '5.0'
                 php:
-                    - '8.0'
```
Comment (Smolevich) [inline .github/workflows/build-ci.yml:44]: New laravel doesn't work with php 8.0?  Why you delete 8.0 from matrix in CI?

Comment (Smolevich) [reply .github/workflows/build-ci.yml:44]: https://github.com/laravel/framework/blob/10.x/composer.json#L18
I suggest to include if condition in ci
If laravel version >= 10.0 then on php 8.0 ci step will be disabled

Comment (divine) [reply .github/workflows/build-ci.yml:44]:
```
- laravel/framework 10.x-dev requires php ^8.1 -> your php version (8.0.27) does not satisfy that requirement.
- illuminate/support 10.x-dev requires php ^8.1 -> your php version (8.0.27) does not satisfy that requirement.
```

**Pull request closed without merging**